cedar-policy-validator/src/err.rs

```diff
@@ -150,11 +150,19 @@ pub enum SchemaError {
     #[error(transparent)]
     #[diagnostic(transparent)]
     UndeclaredActions(#[from] schema_errors::UndeclaredActionsError),
-    /// This error occurs when an undeclared common type appears in entity or context
-    /// attributes.
+    /// This error occurs when we cannot resolve a typename we found in an
+    /// entity or context attribute. (because it refers to an entity type
+    /// or common type that was not declared.)
     #[error(transparent)]
     #[diagnostic(transparent)]
-    UndeclaredCommonTypes(#[from] schema_errors::UndeclaredCommonTypesError),
+    TypeResolution(#[from] schema_errors::TypeResolutionError),
+    /// This error occurs when we cannot resolve a common-type reference.
+    /// Normally, such errors would be `TypeResolution` instead; I think
+    /// this variant should never occur, but not positive, and hard to guarantee
+    /// structurally.
```

Comment:
Elsewhere (e.g. [here](https://github.com/cedar-policy/cedar/blob/b653e6c0627423b24bed50ee664b0302c512f16a/cedar-policy-core/src/parser/err.rs#L228)) we've named these error variants `...InvariantViolation` and included help text asking the user to file as issue.

Reply:
Renamed to `CommonTypeInvariantViolationError` and added appropriate help text